### DISCUSSION TITLE
Radar frequency and pyart.retrieve

### DISCUSSION THREAD

**Discussion by jlpippitt** [ARM-DOE/pyart#1521] [General]
Functions within pyart.retrieve grab frequency data from radar.instrument_parameters["frequency"]["data"][0].  If parameter does not exsist then equations are set to C band.  The problem, S band IRIS radars (NPOL, KPOL) do not contain radar.instrument_parameters["frequency"] information, therefor incorrect equations are applied.  

We used our old rsl_in_idl software and found frequency is in the ray header.
radar.volume.sweep.ray.h
FREQUENCY       FLOAT           2.80968

Can pyART pull frequency from ray header and add radar.instrument_parameters["frequency"] to radar.info() for IRIS radars?

Thank you.

**Comment by mgrover1** [comment]
@jlpippitt - thanks for raising this issue - this would need to be done at the IO level, or you could pass that information through depending on which retrieval you are using. What part of `pyart.retrieve` are you using?

**Reply by jlpippitt** [reply]
Thank you for the quick response.

I was testing pyart.retrieve.hydroclass_semisupervised and pyart.retrieve.est_rain_rate_a.

I set radar.instrument_parameters["frequency"] within my code:
    
    x = ma.array([2.81e+9], dtype=np.float32)
    
     frequency = {'units': 's-1',
    'standard_name': 'radiation_frequency',
    'meta_group': 'instrument_parameters',
    'data': x}  
    
    radar.instrument_parameters["frequency"] = frequency

I was concerned about other users, would it be possible to add at the IO level?

Jason

**Reply by mgrover1** [reply]
Ahh okay - great, thanks! Yes, it would be possible to add this at the IO level. Do you have a sample file we could test with? What file format is this?

**Reply by jlpippitt** [reply]
Raw IRIS NPOL file.

You can download any file from link.
https://pmm-gv.gsfc.nasa.gov/pub/NPOL/Newark/2024/0228/ppi/

Thank you.

**Reply by jlpippitt** [reply]
@mgrover1 - Just a follow up, has the issue been addressed?

Thank you.

**Reply by mgrover1** [reply]
Not yet - I apologize for the delay here

**Reply by mgrover1** [reply]
@jlpippitt - I looked into this a bit, and we have the parameter needed to calculate frequency - wavelength! I just added a PR adding this as a calculated field on IO

https://github.com/ARM-DOE/pyart/pull/1557

NumPy has a utility function for converting wavelength to frequency we can use! I tested it on the file you provided, and it worked as expected! We will ensure this is reviewed and merged.

**Reply by jlpippitt** [reply]
It works, thank you!

**Comment by zssherman** [comment]
Closing this discussion as it was fixed in #1557
